astropy/io/fits/tests/test_core.py

```diff
@@ -1429,13 +1431,9 @@ def test_fix_invalid_extname(self, capsys):
         hdul = fits.HDUList([phdu, ihdu])
         filename = self.temp("temp.fits")
 
-        pytest.raises(
-            fits.VerifyError, hdul.writeto, filename, output_verify="exception"
-        )
```

Comment:
This line is missing a `with`. Not sure what it is supposed to do.

Reply:
it can be used as a function : 
```
When using [pytest.raises()](https://docs.pytest.org/en/latest/reference/reference.html#pytest.raises) as a function, you can use: pytest.raises(Exc, func, match="passed on").match("my pattern").)
```

Reply:
(which I agree is not usual and obvious ;))

Reply:
Oh, oops. Do you want to add it back? If not, I'll get to it when I have time. Sorry!

Reply:
Not sure if it's tested elsewhere, so I will try to keep that on my list. But no problem, it's easy to get confused by this.

Reply:
See https://github.com/astropy/astropy/pull/15252